src/audio/pipewire.c

```diff
@@ -0,0 +1,254 @@
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+
+#include <pipewire/pipewire.h>
+#include <pipewire/thread-loop.h>
+
+#include <spa/param/audio/format-utils.h>
+#include <spa/utils/ringbuffer.h>
+#include <spa/param/props.h>
+
+#include <spd_audio_plugin.h>
+#include "../common/common.h"
+#include <spa/utils/defs.h>
+
+// for the buffer size which will be allocated on the heap, to store the samples produced by speech dispatcher and read by pipewire
+//  this buffer will be used as backing storage by a ring buffer, which will also insure that the threads are syncronised when reading from and writing to the storage area
+#define SAMPLE_BUFFER_SIZE 16 * 1024 // taken from the pipewire ringbuffer example
+
+// speech dispatcher backend entry point, defined in multiple configurations
+#ifdef USE_DLOPEN
+#define SPD_AUDIO_PLUGIN_ENTRY spd_audio_plugin_get
+#else
+#define SPD_AUDIO_PLUGIN_ENTRY spd_pipewire_LTX_spd_audio_plugin_get
+#endif
+
```

Comment:
Please also add the `MSG`/`ERR` macros and put them along the code (and fill set_loglevel), to make bug debunking way easier.

Reply:
I'm not asking to use pipewire functions to do the logging: just copy/paste the macro from another module, they will do the right thing.

Reply:
this is finally fixed in the last commit, closing this as done